elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation;
+
+import com.yahoo.elide.core.DataStore;
+import com.yahoo.elide.core.DataStoreTransaction;
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.core.filter.expression.FilterExpressionVisitor;
+import com.yahoo.elide.core.hibernate.Query;
+import com.yahoo.elide.core.pagination.Pagination;
+import com.yahoo.elide.core.sort.Sorting;
+import com.yahoo.elide.datastores.aggregation.dimension.Dimension;
+import com.yahoo.elide.datastores.aggregation.dimension.TimeDimension;
+import com.yahoo.elide.datastores.aggregation.filter.expression.HavingFilterExpression;
+import com.yahoo.elide.datastores.aggregation.filter.expression.WhereFilterExpression;
+import com.yahoo.elide.datastores.aggregation.metric.Metric;
+
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * A {@link QueryEngine} is an abstraction layer that {@link DataStoreTransaction} leverages to optimize query and send
+ * query, such as loading objects, to a persistent storage.
+ * <p>
+ * The purpose of {@link QueryEngine} is to allow a single {@link DataStore} to utilize multiple query frameworks, such
+ * as Hibernate on SQL and Drill on NoSQL shown below:
+ * <pre>
+ *                           +-----------+                      +------------------+
+ *                           |           |                      |                  |
+ *                           | DataStore |              +-------+ DrillQueryEngine |
+ *                           |           |              |       |                  |
```

Comment:
I don't completely understand this diagram.  I don't understand the DrillQueryEngine and Apache Drill below.  Why are they separate boxes?  

I would probably remove the diagram as it might cause confusion.

Reply:
Sorry I made it confusing. Originally they are separate because `DrillQueryEngine` is a `QueryEngine` implementation, while Apache Drill is out of our control.

 I've updated the diagram; however, if you still don't like it I'm happy to remove it then.